IPython/utils/text.py

```diff
@@ -561,7 +561,13 @@ def _vformat(self, format_string, args, kwargs, used_args, recursion_depth):
 
                 # eval the contents of the field for the object
                 # to be formatted
-                obj = eval(field_name, kwargs)
+                try:
+                    obj = eval(field_name, kwargs)
+                except NameError:
+                    if supressExceptions:
+                       obj = field_name
```

Comment:
I'd wondered where the miracle was going to come in ;)

Is there an easy way to issue a warning, one of those pink messages like when %plab inline colbbers a variable? If the attempt to format the string fails, a user would have a hint about what happened. Currently it's very mysterious and will likely go unnoticed. I guess that's the main reason I raised this issue..

In looking at this, another thing I noticed is a problem with exclamation marks (e.g.: if someone wants '!=' in their expression) - this also silently fails, as ! has special meaning in format strings. There's also the (tiny) chance a someone uses a regex like `a!k:foo`, which doesn't cause Format.parse to raise anything, but returns 'k' as a `conversion` - . Currently this '!k' would disappear! Perhaps this one is "Easy enough to fix" without a miracle (; Perhaps I'll make another PR on that point...